File: src/main/java/org/taktik/icure/logic/ReceiptLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.Receipt;
import org.taktik.icure.entities.embed.ReceiptBlobType;

import java.io.IOException;
import java.util.List;

public interface ReceiptLogic extends EntityPersister<Receipt, String> {
	List<Receipt> listByReference(String ref);

	byte[] getAttachment(String receiptId, String attachmentId) throws IOException;

	void addReceiptAttachment(Receipt receipt, ReceiptBlobType blobType, byte[] payload);
}

File: src/main/kotlin/org/taktik/icure/asynclogic/ReceiptLogic.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */

package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.icure.asyncdao.ReceiptDAO
import org.taktik.icure.entities.Receipt
import org.taktik.icure.entities.embed.ReceiptBlobType
import java.nio.ByteBuffer

interface ReceiptLogic : EntityPersister<Receipt, String> {
    fun listReceiptsByReference(ref: String): Flow<Receipt>
    fun getAttachment(receiptId: String, attachmentId: String): Flow<ByteBuffer>

    suspend fun addReceiptAttachment(receipt: Receipt, blobType: ReceiptBlobType, payload: ByteArray) : Receipt
    fun getGenericDAO(): ReceiptDAO
}
